packages/EasyDoctrine/src/DBAL/Types/CarbonImmutableDateTimeMicrosecondsType.php

```diff
@@ -5,14 +5,18 @@
 namespace EonX\EasyDoctrine\DBAL\Types;
 
 use Carbon\CarbonImmutable;
+use DateTime;
 use DateTimeImmutable;
 use DateTimeInterface;
+use DateTimeZone;
 use Doctrine\DBAL\Platforms\AbstractPlatform;
 use Doctrine\DBAL\Types\ConversionException;
 use Doctrine\DBAL\Types\DateTimeImmutableType;
 
 final class CarbonImmutableDateTimeMicrosecondsType extends DateTimeImmutableType
 {
+    private static ?DateTimeZone $utc = null;
```

Comment:
Do we really need to make it static?

Reply:
Yes, but is it really that helpful? Why don't we use `static` properties everywhere? :)

Reply:
We could just initialise it in the constructor, isn't it?

Reply:
More than that we have started to use Swoole and keeping the state is not a good idea now.

Reply:
I will let @mvlad reply to this. But as of Swoole, it doesn't matter here, as this "state" is kinda constant and useful.

Reply:
I copied this static method from the [Doctrine documentation](https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/cookbook/working-with-datetime.html#:~:text=By%20default%20Doctrine%20assumes%20that,or%20by%20calling%20date_default_timezone_set()%20.). It's needed to optimize the logic.

> We could just initialise it in the constructor, isn't it?

No, `Doctrine\DBAL\Types\Type::__construct` [is final](https://github.com/doctrine/dbal/blob/3.3.x/src/Types/Type.php#L56).

> More than that we have started to use Swoole and keeping the state is not a good idea now.

`DateTimeZone` here is kinda constant as Roman said. This value can't be changed, also `getUtc` is private and can't be used outside the `CarbonImmutableDateTime`. So I don't see even theoretical issues with this logic.